packages/aws-rfdk/lib/deadline/lib/worker-configuration.ts

```diff
@@ -29,6 +29,37 @@ import {
   Version,
 } from './version';
 
+/**
+ * Provider for adding user data scripts
+ */
+export interface IInstanceUserDataProvider {
```

Comment:
The docstrings on this and the `InstanceUserDataProvider` should be better. Approach it from the perspective of someone who is not reading the source code, and only reading the documentation. Will the documentation from these docstrings help you understand what the class/interface is used for? How about how it fits in with the `WorkerInstanceConfiguration` class?

Without knowing the internal workings of the `WorkerInstanceConfiguration` class we want the reader of the documentation to understand where each of the callbacks that are provided by this interface fit into the worker configuration workflow.

Reply:
Docstring for WorkerInstanceConfiguration was updated as well for describe which steps we have.